src/Rules/Functions/UnsetRule.php

```diff
@@ -0,0 +1,84 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\Functions;
+
+use PhpParser\Node;
+use PHPStan\Analyser\Scope;
+use PHPStan\Rules\RuleErrorBuilder;
+use PHPStan\Type\IterableType;
+use PHPStan\Type\VerbosityLevel;
+
+/**
+ * @implements \PHPStan\Rules\Rule<\PhpParser\Node\Stmt\Unset_>
+ */
+class UnsetRule implements \PHPStan\Rules\Rule
+{
+
+	/** @var bool */
+	private $checkMaybeUndefinedVariables;
+
+	public function __construct(bool $checkMaybeUndefinedVariables)
+	{
+		$this->checkMaybeUndefinedVariables = $checkMaybeUndefinedVariables;
+	}
+
+	public function getNodeType(): string
+	{
+		return Node\Stmt\Unset_::class;
+	}
+
+	public function processNode(Node $node, Scope $scope): array
+	{
+		if (!$node instanceof Node\Stmt\Unset_) {
+			return [];
+		}
+
+		/** @var Node\Stmt\Unset_ $node */
+		$functionArguments = $node->vars;
+		$messages = [];
+
+		foreach ($functionArguments as $argument) {
+			$this->canBeUnset($argument, $scope, $messages);
```

Comment:
Please don't pass messages as references. Instead, return an array of strings and merge it with the original array.